src/mono/wasm/runtime/loader/assets.ts

```diff
@@ -600,7 +600,7 @@ function download_resource(asset: AssetEntryInternal): LoadingResource {
         totalResources.add(asset.name!);
         response.response.then(() => {
             if (asset.behavior == "assembly") {
-                loaderHelpers.loadedAssemblies.push(asset.resolvedUrl!);
+                loaderHelpers.loadedAssemblies.push(asset.name);
```

Comment:
Perhaps you also need to push in the name without ".dll" and ".wasm" extension ?

Reply:
No, the name is defined with extension in both ends (lazy loading and asset name)